### DISCUSSION TITLE
GitHub Enterprise Dormant Users report

### DISCUSSION THREAD

**Discussion by ianwilliams1** [community/community#35129] [Enterprise]
The is a report at:
`https://github.com/enterprises/{entname}/settings/compliance`

> **Dormant Users (Beta)**
> Users who have been inactive for 90 days. This report is currently in testing and could be changed or removed at any time.

It has the following fields:

> **created_at** (yyyy-mm-dd hh:mm)
> **id
> login
> role
> last_logged_ip
> 2fa_enabled?**

It would be most helpful if the report included **last_logged_date**. Without that, we can only tell it's been at least 90 days at the time the report was run. Providing that date would allow admins to better cherry-pick "old users" for removal. 

For example, we would like to notify users at the 90 day mark their account will be removed unless the log in, thus resetting their account, then follow up with a removal of accounts now older than a given notice period (say 1 week, so 98 days). The motivation is we have certain support users who are not active committers but may need to do work on an emergency basis on demand and not having their account set up causes support delays.

Alternatively, there is a new "**CSV report  (Beta)**" at:
`https://github.com/enterprises/{entname}/people`

> **GitHub com login
> GitHub com name
> Enterprise server user ids
> GitHub com user
> Enterprise server user
> Visual studio subscription user
> License type
> GitHub com profile
> GitHub com member roles
> GitHub com enterprise role
> GitHub com enterprise roles
> GitHub com verified domain emails
> GitHub com saml name
> GitHub com orgs with pending invites
> GitHub com two factor auth
> Enterprise server primary emails
> Visual studio license status
> Visual studio subscription email
> Total user accounts**

Adding **last_logged_date** to that report may make the other redundant.

**Comment by nearlythere** [comment]
Has there been any response to the suggestion `last_logged_date`? 

This would be really helpful.

**Reply by nearlythere** [reply]
Just one of those little things that is likely never going to happen.

**Comment by jason-dour-e** [comment]
Second the need for last_logged_date.

I'd like to see this report be available for generation via API.

**Comment by Taylerpaul** [comment]
> The is a report at:
> 
> `https://github.com/enterprises/{entname}/settings/compliance`
> 
> 
> 
> > **Dormant Users (Beta)**
> 
> > Users who have been inactive for 90 days. This report is currently in testing and could be changed or removed at any time.
> 
> 
> 
> It has the following fields:
> 
> 
> 
> > **created_at** (yyyy-mm-dd hh:mm)
> 
> > **id
> 
> > login
> 
> > role
> 
> > last_logged_ip
> 
> > 2fa_enabled?**
> 
> 
> 
> It would be most helpful if the report included **last_logged_date**. Without that, we can only tell it's been at least 90 days at the time the report was run. Providing that date would allow admins to better cherry-pick "old users" for removal. 
> 
> 
> 
> For example, we would like to notify users at the 90 day mark their account will be removed unless the log in, thus resetting their account, then follow up with a removal of accounts now older than a given notice period (say 1 week, so 98 days). The motivation is we have certain support users who are not active committers but may need to do work on an emergency basis on demand and not having their account set up causes support delays.
> 
> 
> 
> Alternatively, there is a new "**CSV report  (Beta)**" at:
> 
> `https://github.com/enterprises/{entname}/people`
> 
> 
> 
> 
> 
> > **GitHub com login
> 
> > GitHub com name
> 
> > Enterprise server user ids
> 
> > GitHub com user
> 
> > Enterprise server user
> 
> > Visual studio subscription user
> 
> > License type
> 
> > GitHub com profile
> 
> > GitHub com member roles
> 
> > GitHub com enterprise role
> 
> > GitHub com enterprise roles
> 
> > GitHub com verified domain emails
> 
> > GitHub com saml name
> 
> > GitHub com orgs with pending invites
> 
> > GitHub com two factor auth
> 
> > Enterprise server primary emails
> 
> > Visual studio license status
> 
> > Visual studio subscription email
> 
> > Total user accounts**
> 
> 
> 
> Adding **last_logged_date** to that report may make the other redundant. 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
>

**Comment by timja** [comment]
It would also be good if the `saml_name_id` property was added. As with enterprises users may be onboarded with SCIM and in that case the GitHub username isn't that useful in this report really.

**Comment by smailc** [comment]
Please can the report also include which orgs the user is a member of.
